### PR TITLE
[CURATOR-425] ZooKeeper 3.4.x compatibility mode

### PR DESCRIPTION
**Pull request by Randgalt**
This PR adds ZooKeeper 3.4 compatibility to the master branch. With this, we will no longer need the CURATOR-2.0 branch nor two Curator versions.

### CODE REVIEW

File: curator-test-zk34/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# ZooKeeper 3.4 Compatibility
+
+Apache Curator 4.0 adds best-effort backward compatibility for ZooKeeper 3.4 ensembles and the ZooKeeper 3.4 library/JAR. This module, `curator-test-zk34`, exists to run the standard Curator tests using ZooKeeper 3.4. It does this via Maven. In the curator-test-zk34 pom.xml:
+
+- The Curator modules framework and recipes libraries are included - both main and test JARs - but the ZooKeeper dependency is excluded (otherwise ZooKeeper 3.5.x would be brought in)
+- The curator-test module is included but as version 2.12.0 which brings in ZooKeeper 3.4.8
```
Comment (madrob) [inline curator-test-zk34/README.md]: I'm confused why this is necessary if we're excluding the transitive ZK version anyway?

Comment (Randgalt) [inline curator-test-zk34/README.md]: Which part confuses you?

File: curator-client/src/main/java/org/apache/curator/utils/Compatibility.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    public static void injectSessionExpiration(ZooKeeper zooKeeper)
+    {
+        if ( isZK34() )
+        {
+            InjectSessionExpiration.injectSessionExpiration(zooKeeper);
+        }
+        else
+        {
+            // LOL - this method was proposed by me (JZ) in 2013 for totally unrelated reasons
```
Comment (Randgalt) [inline curator-client/src/main/java/org/apache/curator/utils/Compatibility.java]: I don't appreciate that

**Comment by Randgalt** [comment]
The test seems to work. There a few failures that seem to be related to Jenkins and not this PR. Barring objections I'll merge in the next day or so.

https://builds.apache.org/job/Curator/1069

**Pull request merged**